container-search/src/main/java/com/yahoo/search/Query.java

```diff
@@ -970,10 +971,12 @@ private void copyPropertiesTo(Query clone) {
      * @param create if true this is created if not already set
      * @return the session id of this query, or null if not set and create is false
      */
-    public SessionId getSessionId(boolean create) {
+    public Utf8String getSessionId(String rankProfile, boolean create) {
         if (sessionId == null && create)
             this.sessionId = SessionId.next();
-        return sessionId;
+        return sessionId != null
+                ? new Utf8String(sessionId.toString() + "." + ((rankProfile != null) ? rankProfile : ""))
```

Comment:
I suggest that you pass the parameters to SessionId such that the knowledge about how to mangle them can be encapsulated there.